src/unix/core.c

```diff
@@ -986,6 +986,15 @@ int uv_getrusage(uv_rusage_t* rusage) {
   rusage->ru_nivcsw = usage.ru_nivcsw;
 #endif
 
+  /* Most platforms report ru_maxrss in kilobytes; macOS and Solaris are
+   * the outliers because of course they are.
+   */
+#if defined(__APPLE__)
+  rusage->ru_maxrss /= 1024;                  /* XNU reports bytes. */
```

Comment:
Apparently that's not the case for [iOS](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/getrusage.2.html) which is in kilobytes as well so we should take that into account.

Reply:
Added `&& !TARGET_OS_IPHONE`